jetstream/config.py

```diff
@@ -364,28 +366,32 @@ class MetricDefinition:
     friendly_name: Optional[str] = None
     description: Optional[str] = None
     bigger_is_better: bool = True
+    analysis_basis: Optional[mozanalysis.experiment.AnalysisBasis] = None
 
     def resolve(self, spec: "AnalysisSpec", experiment: ExperimentConfiguration) -> List[Summary]:
         if self.select_expression is None or self.data_source is None:
             # checks if a metric from mozanalysis was referenced
             search = experiment.platform.metrics_module
-            metric = _lookup_name(
+            mozanalysis_metric = _lookup_name(
                 name=self.name,
                 klass=mozanalysis.metrics.Metric,
                 spec=spec,
                 module=search,
                 definitions={},
             )
+            metric = Metric.from_mozanalysis_metric(mozanalysis_metric=mozanalysis_metric)
         else:
             select_expression = _metrics_environment.from_string(self.select_expression).render()
 
-            metric = mozanalysis.metrics.Metric(
+            metric = Metric(
                 name=self.name,
                 data_source=self.data_source.resolve(spec, experiment),
                 select_expr=select_expression,
                 friendly_name=self.friendly_name,
                 description=self.description,
                 bigger_is_better=self.bigger_is_better,
+                analysis_basis=self.analysis_basis
+                or mozanalysis.experiment.AnalysisBasis.ENROLLMENT,
```

Comment:
In general I think it'll be most useful to assume that most metrics should be computed on an exposure basis if an experiment has a useful concept of exposure. Not all of them will.

I think it also sounds confusing if the default basis is different depending on whether or not we've defined something to be true about the experiment.

It sounds like maybe we should back up and figure out what we'd like the configuration files to look like in each case.

We also need to make sure we can communicate the basis of our metrics to the console. Do we want our days_of_use metric to be sometimes computed on an enrollment basis and sometimes computed on an exposure basis, depending on the experiment? Like, maybe, but I dunno. (This _might_ be a reason to revisit the ADR: if we require metrics computed for different bases to have distinct names, that makes it harder to connect the exposure-DOU to the enrollment-DOU, and we will care about both numbers. It might need to be possible to specify bases as ["exposure", "enrollment"], and to introduce a breaking change to the contract of the stats tables so that you need to group by the basis as well. That would probably require a week of data science feedback time but I don't think that's intolerable.)

Reply:
I added some updates to the ADR in https://github.com/mozilla/jetstream/pull/724/

Reply:
For now I implemented option 1, which seems more convenient when working with the statistics results even if it might break the contract. I'm not sure if there are data scientists that actually work with the statistics tables directly that would be impacted or if it would only affect Experimenter console?

Reply:
Su has some redash dashboards; I'll set up a PR to DTMO and circulate it as the notification.